contracts/Lottery.sol

```diff
@@ -19,72 +20,103 @@ contract Lottery {
         uint lotteryId;
         uint price;
-        address payable ticketOwner;
+        address ticketOwner;
     }
     
 
     address private owner;
-    uint32 private totalAmount;
-    uint32 private ticketPrice;
-    Player[] public winner;
+    LotteryState private state;
+    uint private ticketPrice;
+    Player private winner;
     
-    mapping (address => Player) playerMap;
     Player[] players;
     LotteryTicket[] tickets;
-
+    
+    mapping (address => Player) playerMap;
+    mapping(address => LotteryTicket[]) playerLotteryTicketsMapping;
+    
+    //events
+    event NewUser(address user);
+    event BuyTicket(address buyer, uint ticketsCount, uint totalAnount);
+    
+    event LotteryWinner(address winner, uint amount);
+    event LotteryStatusUpdate(address player, string status);
+    event LotteryStatusUpdateWithMessage(address player, string status, string message);
+    
+    // This will be used if lottery owner does premature termination of lottery and in this case amount should be refunded to each player
+    event RefundLottery(address player, uint amount); 
+    
     constructor () public {
         state = LotteryState.NotStarted;
         owner = msg.sender;
     }
-
-    function registerPlayer(string memory _name, string memory _emailId, string memory  _phoneNumber) public payable returns (string memory result) {
+    
+    function registerPlayer(string memory _name, string memory _emailId, string memory  _phoneNumber) public {
         //Owner shouldn't be player
-        require(owner != msg.sender);
-        require((state != LotteryState.NotStarted) && (state != LotteryState.Finished), "Player can't register while lottery has started or finished.");
+        require(owner != msg.sender, "Lottery owner can't register as player.");
+        require(state == LotteryState.Started, "Player can't register while lottery has started or finished.");
+        require(!playerMap[msg.sender].isExist, "User already registered as player");
+        
+        Player memory _player = Player(msg.sender, _name, _emailId, _phoneNumber,true);
+        playerMap[msg.sender] = _player;
+        players.push(_player);
         
-        result = "FAIL !!!";
-        if(!playerMap[msg.sender].isExist ) {
-            Player memory _player = Player(msg.sender, _name, _emailId, _phoneNumber,true);
-            playerMap[msg.sender] = _player;
-            players.push(_player);
-            result = "Registered Successfully !!!";
-        }
-
-        return(result);
         // new event to be send to owner
+        emit NewUser(msg.sender);
+
     }
     
-    function startLottery(uint32 _ticketPrice) public{
+    function startLottery(uint _ticketPrice) public{
         require (owner == msg.sender, "Only owner can open lottery process.");
-        state = LotteryState.Started;
         ticketPrice = _ticketPrice;
+        state = LotteryState.Started;
     }
-
-    function stopLottery() public{
```

Comment:
Why has this function been removed?

Reply:
There was no clear purpose of this function. Lottery should be finished after choosing winner.

Reply:
As stated above, We should keep this function.